aggregator/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/DailyEmailAggregationJob.java

```diff
@@ -10,6 +12,7 @@
 import io.vertx.core.Vertx;
 import io.vertx.kafka.client.producer.KafkaProducer;
 import io.vertx.kafka.client.producer.KafkaProducerRecord;
```

Comment:
Is there any reason why [quarkus-smallrye-reactive-messaging](https://quarkus.io/guides/kafka) isn't used here?

Reply:
Yep. We tried to use it in cloudwatch-logging-kafka already and neither Heiko, me nor Peter Palaga got it up and running.
Feel free if you want to give it a try.

Reply:
What was the problem exactly?

Having direct dependencies on `vertx-kafka-client` and `kafka-clients` isn't recommended by the Quarkus team. Depending on `quarkus-kafka-client` would be a bit better but still not the best way to go. Ideally we should depend on `quarkus-smallrye-reactive-messaging`.

Reply:
I can't remember what exactly the problem was. I remember we spend some time on trying to get it up and running and used another dependency in the end.

Why isn't it recommended?

Reply:
I can't remember what exactly the problem was. I remember we spend some time on trying to get it up and running and used another dependency in the end.

Why isn't it recommended?

Will try to use it later again, but won't spend much time with it. Timeboxed for 2 hours. If it's working then, it's fine. If not, I will put it on my list and do it later.

Reply:
Because we won't benefit from any of the following things if we don't use the Quarkus extensions:
- Better integration between Kafka and Quarkus core and other extensions. I had a super quick look at `quarkus-kafka-client`, it contains integration code in many areas like startup/shutdown, serialization, health, dev mode, security... `quarkus-smallrye-reactive-messaging` certainly contains even more.
- Build time optimizations (faster boot, lower resources consumption) which mean better performances.
- Native mode compatibility (which is probably not important now). Kafka won't work in native mode without the Quarkus extensions.
- More importantly, support from the Quarkus team when we have a question or something goes wrong.

And I'm sure I am forgetting other important reasons 😄

Reply:
> Will try to use it later again, but won't spend much time with it. Timeboxed for 2 hours. If it's working then, it's fine. If not, I will put it on my list and do it later.

It can be fixed later indeed.

Reply:
Btw notifications-gw sends to Kafka like this https://github.com/RedHatInsights/notifications-gw/blob/main/src/main/java/com/redhat/cloud/notifications/GwResource.java#L52 and https://github.com/RedHatInsights/notifications-gw/blob/main/src/main/java/com/redhat/cloud/notifications/GwResource.java#L95

Reply:
My expected timebox wasn't that bad.